purchase_product_packaging_container_deposit/tests/test_purchase_product_packaging_container_deposit.py

```diff
@@ -0,0 +1,283 @@
+        self.assertEqual(box_line.product_qty, 20)
+
+    def test_purchase_product_packaging_container_deposit_quantities_case6(self):
+        """
+        Case 6: Product A | qty = 280 on confirmed order.
+                Product A | qty_received = 280. Result:
+                Received 1 Pallet
+                Received 11 Boxes
+
+        """
+        self.env["purchase.order.line"].create(
+            [
+                {
+                    "order_id": self.purchase_order.id,
+                    "name": self.product_a.name,
+                    "product_id": self.product_a.id,
+                    "product_qty": 280,
+                },
+            ]
+        )
+        self.purchase_order.button_confirm()
+
+        pick = self.purchase_order.picking_ids
+        pick.move_ids.write({"quantity_done": 280})
+        pick.button_validate()
+
+        pallet_line = self.purchase_order.order_line.filtered(
+            lambda ol: ol.product_id == self.pallet
+        )
+        box_line = self.purchase_order.order_line.filtered(
+            lambda ol: ol.product_id == self.box
+        )
+        self.assertEqual(pallet_line.qty_received, 1)
+        self.assertEqual(box_line.qty_received, 11)
+
+    def test_purchase_product_packaging_container_deposit_quantities_case7(self):
+        """
+        Case 7.1: Product A | qty = 280 on confirmed order.
+                Product A | Partial shipment (qty_received = 140). Result:
+                Received 140 // 280 = 0 Pallet
+                Received 140 // 24 = 5 Boxes
+        Case 7.2: Product A | Increase received quantity (qty_received = 200). Result:
+
+                Received 200 // 280 = 0 Pallet
+                Received 200 // 24 = 5 Boxes
+        """
+        self.env["purchase.order.line"].create(
+            [
+                {
+                    "order_id": self.purchase_order.id,
+                    "name": self.product_a.name,
+                    "product_id": self.product_a.id,
+                    "product_qty": 280,
+                },
+            ]
+        )
+        self.purchase_order.button_confirm()
+
+        pick = self.purchase_order.picking_ids
+        pick.move_ids.write({"quantity_done": 140})
+        wiz_act = pick.button_validate()
+        wiz = Form(
+            self.env[wiz_act["res_model"]].with_context(**wiz_act["context"])
+        ).save()
+        wiz.process()
+
+        pallet_line = self.purchase_order.order_line.filtered(
+            lambda ol: ol.product_id == self.pallet
+        )
+        box_line = self.purchase_order.order_line.filtered(
+            lambda ol: ol.product_id == self.box
+        )
+
+        (pallet_line | box_line).invalidate_cache()
+        self.assertEqual(pallet_line.qty_received, 0)
+        self.assertEqual(box_line.qty_received, 5)
+
+        self.purchase_order.order_line[0].qty_received = 200
+        self.assertEqual(pallet_line.qty_received, 0)
+        self.assertEqual(box_line.qty_received, 8)
```

Comment:
You are not testing the form here. Can you add the same test as in the sale module?

Reply:
The Form unit test was added.